### PR TITLE
Add sensitive to certain file creation calls

### PR DESCRIPTION
**Pull request by martinisoft**
This will help improve security by not logging sensitive information (specifically SSL private keys, not really the public cert). I believe I've gotten all the pertinent file calls that could reveal sensitive information, but feel free to correct me and I'll update the PR.

### CODE REVIEW

**Comment by jhmartin** [comment]
The sensitive attribute arrived in Chef 11.12, so will adding this here break older Chef's (if they are supported?)

**Comment by martinisoft** [comment]
@jhmartin Looks like it actually came in around [11.14](https://github.com/chef/chef/commit/cee94f52d52806885fbd63d63addc3708b25f409) so you have a good point there.

**Comment by martinisoft** [comment]
@portertech Which target versions of Chef are you aiming for with this cookbook?

**Comment by portertech** [comment]
@martinisoft maybe throw a conditional into those lines?

**Comment by martinisoft** [comment]
@portertech https://github.com/martinisoft/sensu-chef/commit/b1c9d08fd17c7b485b70fc79c1439351f7ed01ae should address the issue.

File: recipes/default.rb [hunk 1]
```diff
@@ -80,6 +80,7 @@
     owner node.sensu.admin_user
     group node.sensu.group
     mode 0640
+    sensitive true if Chef::Resource::ChefGem.instance_methods(false).include?(:sensitive)
```
Comment (cbandy) [inline recipes/default.rb]: Does [`respond_to?`](http://ruby-doc.org/core-1.9.3/Object.html#method-i-respond_to-3F) work instead?

```ruby
sensitive true if respond_to? :sensitive
```

Comment (martinisoft) [inline recipes/default.rb]: @cbandy I think there were some issues with this and previous versions of Chef due to how the compile and run phases work. A similar patch was attempted when the `compile_time` attribute was introduced and the `respond_to?` method did not work.

**Comment by portertech** [comment]
Let's do this! :+1:

File: recipes/default.rb (hunk 1 above)
Comment (scalp42) [inline recipes/default.rb]: see https://github.com/opscode-cookbooks/aws/pull/110 and http://jtimberman.housepub.org/blog/2015/03/20/chef-gem-compile-time-compatibility/

**Pull request merged**